### DISCUSSION TITLE
How do I increase audio inputs on main page

### DISCUSSION THREAD

**Discussion by vegedb** [staxrip/staxrip#648] [General]
I'd like to have all my audio be converted to opus.
![image](https://user-images.githubusercontent.com/72070424/113486925-b16d0400-94e7-11eb-8950-7066dfcdc9c7.png)

But there are only 2 options here. This is strictly for **batch** files as I know I can change it manually for single files using "container options". How do I increase this to 4 or more?

**Comment by ghost** [comment]
@vegedb I'm afraid it is a limitation in Staxrip, since the feature of more than 2 audio tracks was not in the initial design of Staxrip...
Maybe @stax76 can confirm this...

**Reply by vegedb** [reply]
@44vince44 
Its possible to do it on individual files though, I bet this is an easy fix or there's already an option that I'm not aware of.

**Reply by JJKylee** [reply]
I can't think of any, either.

> I bet this is an easy fix...

Well, let's see.

But I'm afraid it's not an easy task for StaxRip due to its design limit. (But I also hope I'm wrong.) 🤔

As a matter of fact, I can't think of any (FFmpeg) GUI program that enables loading _all or multiple_ audio tracks automatically.
Only a CLI program like FFmpeg provides that kind of feature, IMHO.
Correct me if I'm wrong. 😉

**Reply by stax76** [reply]
I wouldn't change the main dialog. Supporting all tracks for batch processing was probably requested before, not often. I don't think I tried building it, because it wasn't requested often enough, and it didn't seem to be important enough, or maybe I thought it was too difficult or hadn't a good idea how to build it. I'm not aware of something that would block it however.

**Comment by jessielw** [comment]
Solution, encode all video with staxrip, encode all audio with ffmpeg, then batch mux the files together with ffmpeg or mkvmerge or something like that.

**Reply by JJKylee** [reply]
> ...encode all audio with ffmpeg...

Well, if the codec is AAC, I would disagree with this suggestion, but since what @vegedb wants is OPUS, this makes sense. 😅

To top it off, if you want the video codec to be HEVC or AVC and would like to encode with `x265` or `x264` _without_ any AVS/VS-only filters, batch processing with `ffmpeg` alone is not a bad idea. 😁

**Reply by jessielw** [reply]
Yeah I personally would use QAAC. I do a lot of TV series and I use a lot of batch scripts. Staxrip to just do the video for TV series generally.
